clustered/client/src/test/java/org/ehcache/clustered/client/docs/ClusteredLoaderWriter.java

```diff
@@ -0,0 +1,125 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.client.docs;
+
+import org.ehcache.Cache;
+import org.ehcache.PersistentCacheManager;
+import org.ehcache.clustered.client.config.builders.ClusteredResourcePoolBuilder;
+import org.ehcache.clustered.client.config.builders.ClusteringServiceConfigurationBuilder;
+import org.ehcache.clustered.client.internal.UnitTestConnectionService;
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.config.builders.CacheConfigurationBuilder;
+import org.ehcache.config.builders.CacheManagerBuilder;
+import org.ehcache.config.builders.ResourcePoolsBuilder;
+import org.ehcache.config.builders.WriteBehindConfigurationBuilder;
+import org.ehcache.config.units.MemoryUnit;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Samples demonstrating clustered cache with loader-writers & write-behind.
+ */
+public class ClusteredLoaderWriter {
+
+  private static final String CLUSTER_URI = "terracotta://localhost/my-application";
+
+  @Before
+  public void resetPassthroughServer() throws Exception {
+    UnitTestConnectionService.add(CLUSTER_URI,
+      new UnitTestConnectionService.PassthroughServerBuilder()
+        .resource("primary-server-resource", 8, MemoryUnit.MB)
+        .resource("secondary-server-resource", 4, MemoryUnit.MB)
+        .build());
+  }
+
+  @After
+  public void removePassthroughServer() throws Exception {
+    UnitTestConnectionService.remove("terracotta://localhost/my-application");
+  }
+
+  @Test
+  public void clusteredWriteThroughCache() throws Exception {
+    CacheManagerBuilder<PersistentCacheManager> clusteredCacheManagerBuilder =
+      CacheManagerBuilder.newCacheManagerBuilder()
+        .with(ClusteringServiceConfigurationBuilder.cluster(URI.create(CLUSTER_URI)).autoCreate()
+          .defaultServerResource("primary-server-resource"));
+
+    // tag::clusteredWriteThroughCache[]
+    try (PersistentCacheManager cacheManager = clusteredCacheManagerBuilder.build(true)) {
+      Map<Long, String> sor = new HashMap<>();
+      sor.put(4L, "Known Four");
+      SampleLoaderWriter<Long, String> loaderWriter = new SampleLoaderWriter<>(sor); // <1>
+
+      CacheConfiguration<Long, String> configuration = CacheConfigurationBuilder
+        .newCacheConfigurationBuilder(Long.class, String.class, ResourcePoolsBuilder.newResourcePoolsBuilder()
+          .heap(2, MemoryUnit.MB)
```

Comment:
I think I'd prefer it if we stuck to count based sizes for heap resources in samples.